drivers/flash/flash_simulator.c

```diff
@@ -372,46 +378,58 @@ static int flash_mock_init(const struct device *dev)
 	struct stat f_stat;
 	int rc;
 
-	if (flash_file_path == NULL) {
-		flash_file_path = default_flash_file_path;
-	}
+	ARG_UNUSED(dev);
 
-	flash_fd = open(flash_file_path, O_RDWR | O_CREAT, (mode_t)0600);
-	if (flash_fd == -1) {
-		posix_print_warning("Failed to open flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
-	}
+	if (flash_on_ram == true) {
+		mock_flash = (uint8_t *)malloc(FLASH_SIMULATOR_FLASH_SIZE);
+		if (mock_flash == NULL) {
+			posix_print_warning("Could not allocate flash in Linux process heap %s\n",
+					    strerror(errno));
+			return -EIO;
+		}
+	} else {
 
-	rc = fstat(flash_fd, &f_stat);
-	if (rc) {
-		posix_print_warning("Failed to get status of flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
-	}
+		if (flash_file_path == NULL) {
+			flash_file_path = default_flash_file_path;
+		}
 
-	if (ftruncate(flash_fd, FLASH_SIMULATOR_FLASH_SIZE) == -1) {
-		posix_print_warning("Failed to resize flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
-	}
+		flash_fd = open(flash_file_path, O_RDWR | O_CREAT, (mode_t)0600);
+		if (flash_fd == -1) {
+			posix_print_warning("Failed to open flash device file "
+					"%s: %s\n",
+					flash_file_path, strerror(errno));
+			return -EIO;
+		}
+
+		rc = fstat(flash_fd, &f_stat);
+		if (rc) {
+			posix_print_warning("Failed to get status of flash device file "
+					"%s: %s\n",
+					flash_file_path, strerror(errno));
+			return -EIO;
+		}
+
+		if (ftruncate(flash_fd, FLASH_SIMULATOR_FLASH_SIZE) == -1) {
+			posix_print_warning("Failed to resize flash device file "
+					"%s: %s\n",
+					flash_file_path, strerror(errno));
+			return -EIO;
+		}
 
-	mock_flash = mmap(NULL, FLASH_SIMULATOR_FLASH_SIZE,
-			  PROT_WRITE | PROT_READ, MAP_SHARED, flash_fd, 0);
-	if (mock_flash == MAP_FAILED) {
-		posix_print_warning("Failed to mmap flash device file "
-				    "%s: %s\n",
-				    flash_file_path, strerror(errno));
-		return -EIO;
+		mock_flash = mmap(NULL, FLASH_SIMULATOR_FLASH_SIZE,
+				PROT_WRITE | PROT_READ, MAP_SHARED, flash_fd, 0);
+		if (mock_flash == MAP_FAILED) {
+			posix_print_warning("Failed to mmap flash device file "
+					"%s: %s\n",
+					flash_file_path, strerror(errno));
+			return -EIO;
+		}
 	}
 
-	if (f_stat.st_size == 0) {
-		/* erase the memory unit by pulling all bits to one */
+	if ((flash_erase_at_start == true) || (flash_on_ram == true) || (f_stat.st_size == 0)) {
```

Comment:
So erase happens by default for RAM?

Reply:
Yep. Just like when the flash simulator runs in other archs. I will add a comment in the command line option.